test/e2e/buildfarm_test.go

```diff
@@ -0,0 +1,219 @@
+package integration
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/containers/common/pkg/config"
+	. "github.com/containers/podman/v4/test/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+)
+
+func setupContainersConfWithSystemConnections() {
+	// make sure connections are not written to real user config on host
+	file := filepath.Join(podmanTest.TempDir, "containersconf")
+	f, err := os.Create(file)
+	Expect(err).ToNot(HaveOccurred())
+	connections := `
+[engine]
```

Comment:
I think we should create a new `buildfarm` table to keep a clear separation of concerns.

Reply:
Can you please elaborate, do you mean a table for the various test cases?

Reply:
I mean that it should not be part of the `[engine]` table but a new `[buildfarm]` table.

Ideally, it should not be part of `containers.conf` at all but a proper database or a dedicated predictable path for root and rootless.  `machine` and `conenction` are (ab)using containers.conf already as a database but there is growing desire to change that, see https://github.com/containers/common/pull/1566.  `containers.conf` is a config file for users and we somehow abuse it as a database.

Cc: @rhatdan @Luap99

Reply:
Ah I see, thanks! If we are moving service destinations to a `connections.conf` file, I think it would make sens to use the same file for Farms as well as Farms are closely tied with service destinations. A Farm points to a list of service destinations that will be used to farm out the builds to. WDYT?

Reply:
Let's continue the conversation in https://github.com/containers/common/pull/1586.  TBH, I am not entirely sure how to proceed and don't want to block the buildfarm work on sins of the past :^)  I was strongly against using containers.conf as a DB but made my peace.

Reply:
I created a new table for Farms in https://github.com/containers/common/pull/1586. Fixed